django/middleware/csrf.py

```diff
@@ -26,7 +26,9 @@
 REASON_MALFORMED_REFERER = "Referer checking failed - Referer is malformed."
 REASON_INSECURE_REFERER = "Referer checking failed - Referer is insecure while host is secure."
 
-CSRF_KEY_LENGTH = 32
+CSRF_NONCE_LENGTH = 32
+CSRF_TOKEN_LENGTH = 2 * CSRF_NONCE_LENGTH
+CSRF_ALLOWED_CHARS = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789'
```

Comment:
string.letters doesn't exist in Python 3 nor in six, so this required an enhancement to our own six, but, done.